borg/_hashindex.c

```diff
@@ -473,7 +473,7 @@ hashindex_add(HashIndex *index, const void *key, int32_t *other_values)
     if(my_values == NULL) {
         hashindex_set(index, key, other_values);
     } else {
-        *my_values += *other_values;
+        my_values[0] = _htole32(_le32toh(my_values[0]) + _le32toh(other_values[0]));
```

Comment:
On BE it would add LE numbers (with a BE-add) and interpret the result as LE. In some cases that works, but as soon as a carry crosses a byte-border it breaks. This might even create smaller results, or overflow earlier. If it makes smaller results = refcounts data might have been deleted by delete/prune.

Now this code should work correctly everywhere. For LE these macros are a no-op. For BE the values (LE) are converted to BE, added with BE-add, and converted back to LE.

(None of this resides in the Repository, this is only used for the Cache. Deleting caches -- they are automatically recreated -- will purge any incorrect refcounts that might linger around. Use extract --dry-run on all archives to verify that no data has been lost).